students/paul_c/session07/html_render.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+
+    tag = 'html'
+
+    def __init__(self, content=None, **kwargs):
+        self.attributes = kwargs
+        self.contents = []
+        if content is not None:
+            self.contents = [content]
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+    def render(self, out_file):
+        open_tag = ["<{}".format(self.tag)]
+        for key, value in self.attributes.items():
+            open_tag.append(' ' + key + '="' + str(value) + '"')
```

Comment:
This works, but you should add the closing ">" to the tag, after you have added all attributes (if any):
```
        open_tag.append(">\n")
```